lib/parse.js

```diff
@@ -237,14 +237,36 @@ var normalizeParseOptions = function normalizeParseOptions(opts) {
     };
 };
 
+var decodeObject = function (obj, opts) {
+    if (opts.decoder === defaults.decoder) {
+        return obj;
+    }
```

Comment:
i think this optimization is actually incorrect, because the default decoder does actual work.
```suggestion
```